server/app/services/grid_service_deployer.rb

```diff
@@ -57,7 +57,9 @@ def deploy
       self.deploy_service_instance(total_instances, deploy_futures, instance_number, deploy_rev)
       sleep 0.1
     end
-    deploy_futures.select{|f| !f.ready?}.each{|f| f.value }
```

Comment:
I think this never noticed instance deploy errors from the tail-end instance deploy futures in cases where `deploy_service_instance` didn't wait for them.